### PR TITLE
chore(auth): Change API tests to use postgres instead of sqlite

### PR DESCRIPTION
**Pull request by eirikurn**
## What

See title

## Why

- There are subtle differences between sqlite and postgres which can cause production errors to not be caught in CI.
- We want to use postgres features which are not supported in sqlite.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by Herdismaria — APPROVED** [review]

**Review by FridrikMProgramm — APPROVED** [review]
LGTM

File: libs/auth-api-lib/src/lib/idp-provider/models/idp-provider.model.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ export class IdpProvider extends Model {
   helptext!: string
 
   @Column({
-    type: DataType.NUMBER,
+    type: DataType.INTEGER,
```
Comment (baering) [inline libs/auth-api-lib/src/lib/idp-provider/models/idp-provider.model.ts:40]: No migration required?

File: libs/auth-api-lib/src/lib/clients/models/client.model.ts [hunk 2]
```diff
@@ -357,7 +357,7 @@ export class Client extends Model {
   backChannelLogoutUri?: string
 
   @Column({
-    type: DataType.STRING,
+    type: DataType.BOOLEAN,
```
Comment (baering) [inline libs/auth-api-lib/src/lib/clients/models/client.model.ts:360]: No migration required?

File: libs/auth-api-lib/src/lib/resources/models/api-scope-group.model.ts [hunk 3]
```diff
@@ -63,7 +63,7 @@ export class ApiScopeGroup extends Model<
   description!: string
 
   @Column({
-    type: DataType.NUMBER,
+    type: DataType.INTEGER,
```
Comment (baering) [inline libs/auth-api-lib/src/lib/resources/models/api-scope-group.model.ts:66]: No migration required?

File: libs/auth-api-lib/src/lib/resources/models/api-scope-user-access.model.ts [hunk 4]
```diff
@@ -28,9 +28,8 @@ export class ApiScopeUserAccess extends Model {
 
   @PrimaryKey
   @Column({
-    type: DataType.BOOLEAN,
+    type: DataType.STRING,
```
Comment (baering) [inline libs/auth-api-lib/src/lib/resources/models/api-scope-user-access.model.ts:31]: No migration required?

File: libs/auth-api-lib/src/lib/resources/models/api-scope.model.ts [hunk 5]
```diff
@@ -111,7 +111,7 @@ export class ApiScope extends Model<ModelAttributes, CreationAttributes> {
   domainName!: string
 
   @Column({
-    type: DataType.NUMBER,
+    type: DataType.INTEGER,
```
Comment (baering) [inline libs/auth-api-lib/src/lib/resources/models/api-scope.model.ts:114]: No migration required?

**Review by baering — APPROVED** [review]

File: libs/auth-api-lib/src/lib/idp-provider/models/idp-provider.model.ts (hunk 1 above)
Comment (eirikurn) [reply libs/auth-api-lib/src/lib/idp-provider/models/idp-provider.model.ts:40]: These decorator types are just used in jest tests and were out of sync with the postgres db schema. 👍

**Pull request merged**